File: app/src/main/java/jp/juggler/subwaytooter/MyFirebaseMessagingService.java
package jp.juggler.subwaytooter;

import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.SystemClock;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import java.util.Map;

import jp.juggler.subwaytooter.util.LogCategory;

public class MyFirebaseMessagingService extends FirebaseMessagingService {
	static final LogCategory log = new LogCategory( "MyFirebaseMessagingService" );
	
	@Override public void onCreate(){
		super.onCreate();
	}
	
	@Override public void onMessageReceived( RemoteMessage remoteMessage ){

		super.onMessageReceived( remoteMessage );
		
		String tag = null;
		Map< String, String > data = remoteMessage.getData();
		if( data != null ){
			for( Map.Entry< String, String > entry : data.entrySet() ){
				log.d( "onMessageReceived: %s=%s", entry.getKey(), entry.getValue() );
				
				if( "notification_tag".equals( entry.getKey() ) ){
					tag = entry.getValue();
				}
			}
		}

		Context context = getApplicationContext();
		Intent intent = new Intent( context,PollingForegrounder.class);
		if(tag != null ) intent.putExtra(PollingWorker.EXTRA_TAG,tag);
		if( Build.VERSION.SDK_INT >= 26 ){
			context.startForegroundService( intent );
		}else{
			context.startService(intent);
		}
	}
	
}

File: app/src/main/java/jp/juggler/subwaytooter/MyFirebaseMessagingService.kt
package jp.juggler.subwaytooter
//
//import com.google.firebase.messaging.FirebaseMessagingService
//import com.google.firebase.messaging.RemoteMessage
//import jp.juggler.subwaytooter.api.entity.Acct
//import jp.juggler.subwaytooter.notification.PollingChecker
//import jp.juggler.subwaytooter.notification.restartAllWorker
//import jp.juggler.subwaytooter.pref.PrefDevice
//import jp.juggler.subwaytooter.pref.prefDevice
//import jp.juggler.subwaytooter.table.SavedAccount
//import jp.juggler.subwaytooter.table.apiNotificationCache
//import jp.juggler.subwaytooter.table.apiSavedAccount
//import jp.juggler.util.log.LogCategory
//import kotlinx.coroutines.runBlocking
//import java.util.*
//
//class MyFirebaseMessagingService : FirebaseMessagingService() {
//
//    companion object {
//        internal val log = LogCategory("MyFirebaseMessagingService")
//
//        private val pushMessageStatus = LinkedList<String>()
//
//        // Pushメッセージが処理済みか調べる
//        private fun isDuplicateMessage(messageId: String) =
//            synchronized(pushMessageStatus) {
//                when (pushMessageStatus.contains(messageId)) {
//                    true -> true
//                    else -> {
//                        pushMessageStatus.addFirst(messageId)
//                        while (pushMessageStatus.size > 100) {
//                            pushMessageStatus.removeLastCompat()
//                        }
//                        false
//                    }
//                }
//            }
//    }
//
//    override fun onNewToken(token: String) {
//        try {
//            log.w("onTokenRefresh: token=$token")
//            prefDevice.device
//            pollingWorker2IntervalPrefDevice.from(this).edit().putString(PrefDevice.KEY_DEVICE_TOKEN, token).apply()
//            restartAllWorker(this)
//        } catch (ex: Throwable) {
//            log.e(ex, "onNewToken failed")
//        }
//    }
//
//    override fun onMessageReceived(remoteMessage: RemoteMessage) {
//        val messageId = remoteMessage.messageId ?: return
//        if (isDuplicateMessage(messageId)) return
//
//        val accounts = ArrayList<SavedAccount>()
//        for ((key, value) in remoteMessage.data) {
//            log.w("onMessageReceived: $key=$value")
//            when (key) {
////                "notification_tag" -> {
////                    apiSavedAccount.(context, value).forEach { sa ->
////                        apiNotificationCache.resetLastLoad(sa.db_id)
////                        accounts.add(sa)
////                    }
////                }
//                "acct" -> {
//                    apiSavedAccount.loadAccountByAcct(Acct.parse(value))?.let { sa ->
//                        apiNotificationCache.resetLastLoad(sa.db_id)
//                        accounts.add(sa)
//                    }
//                }
//            }
//        }
//
//        if (accounts.isEmpty()) {
//            // タグにマッチする情報がなかった場合、全部読み直す
//            apiNotificationCache.resetLastLoad()
//            accounts.addAll(apiSavedAccount.loadAccountList())
//        }
//
//        log.i("accounts.size=${accounts.size} thred=${Thread.currentThread().name}")
//        runBlocking {
//            accounts.forEach {
//                check(it.db_id)
//            }
//        }
//    }
//
//    private suspend fun check(accountDbId: Long) {
//        try {
//            PollingChecker(
//                context = this,
//                accountDbId = accountDbId
//            ).check { a, s ->
//                val text = "[${a.acct.pretty}]${s.desc}"
//                log.i(text)
//            }
//        } catch (ex: Throwable) {
//            log.e(ex, "check failed. accountDbId=$accountDbId")
//        }
//    }
//}
